packages/next/src/build/create-compiler-aliases.ts

```diff
@@ -237,6 +237,7 @@ export function createAppRouterApiAliases(isServerOnlyLayer: boolean) {
   const mapping: Record<string, string> = {
     head: 'next/dist/client/components/noop-head',
     dynamic: 'next/dist/api/app-dynamic',
+    link: 'next/dist/client/app-dir/link',
```

Comment:
I considered moving `noop-head` into this folder as well since we alias it for a similar reason, but I wasn't following the distinction for when we put something in `client` vs `client/components` so I'm ignoring this for now.